2-dom-manipulation/form.js

```diff
@@ -17,37 +16,51 @@ function validateForm(event) {
         courses: document.querySelectorAll('input[type="checkbox"]'),
         message: document.querySelector('#texta')
     };
-    printInHTML(form.name, form.email, form.phone, form.courses, form.message);
+    deleteAllInformationParagraphs();
+    let name = validateName(form.name);
+    let email = validateEmail(form.email);
+    let phone = validatePhoneNumber(form.phone);
+    let courses = Array.isArray(validateCourses(form.courses));
+    if (name && email && phone && courses) {
+        printInHTML(form.name.value);
+        printInHTML(form.email.value);
+        printInHTML(form.phone.value);
+        printInHTML(validateCourses(form.courses));
```

Comment:
it's necessary to call validateCourses twice? (please let me know).

Reply:
Also you can use `for…in` to iterate over the properties of an object (the object keys):

Reply:
I had to call the function twice because if I call the functions into the if statement, it trunks if for example no course is selected, and all the other elements are correct.